gcp_variant_transforms/beam_io/vcfio.py

```diff
@@ -207,6 +209,145 @@ def __repr__(self):
+    # Set end in info if this is a non-variant.
+    if (variant.start and variant.reference_bases and
+        variant.start + len(variant.reference_bases) < variant.end):
+      info_list.append('END=%d' % variant.end)
+    # Set all other fields of info.
+    for k, v in variant.info.iteritems():
+      if v.field_count == '1':
+        info_list.append('%s=%s' % (k, str(v.data)))
+      else:
+        info_list.append('%s=%s' % (k, ','.join([str(x) for x in v.data])))
+    return ';'.join(info_list)
+
+  def _get_variant_format(self, variant):
+    format_keys = ['GT']
+    if variant.calls:
+      # If any calls have a set phaseset that is not `DEFAULT_PHASESET_VALUE`,
+      # the key will be added to the format field.
+      for call in variant.calls:
+        if call.phaseset and call.phaseset != DEFAULT_PHASESET_VALUE:
+          format_keys.append('PS')
+          break
+      format_keys.extend([k for k in sorted(variant.calls[0].info)])
+    return ':'.join(format_keys)
+
+  def _get_variant_calls(self, variant):
+    calls_values = []
+    alternate_phaseset = False
+
+    for call in variant.calls:
+      if call.phaseset and call.phaseset != DEFAULT_PHASESET_VALUE:
+        alternate_phaseset = True
+        break
+
+    for call in variant.calls:
+      # Set GT field.
+      genotype_field = MISSING_FIELD_VALUE
+      if call.genotype != [MISSING_GENOTYPE_VALUE]:
+        phase_char = '/' if call.phaseset is None else '|'
+        genotype_field = phase_char.join([str(x) for x in call.genotype])
+      call_info_fields = [genotype_field]
+      # Set PS field if there is a set phaseset that is not
+      # `DEFAULT_PHASET_VALUE`.
+      if alternate_phaseset:
+        if call.phaseset and call.phaseset != DEFAULT_PHASESET_VALUE:
+          call_info_fields.append(call.phaseset)
+        else:
+          call_info_fields.append(MISSING_FIELD_VALUE)
+      # Set all other fields of the call info.
+      for k in sorted(call.info):
+        if call.info[k]:
+          call_info_fields.append(str(call.info[k]))
+        else:
+          call_info_fields.append(MISSING_FIELD_VALUE)
+
+      calls_values.append(':'.join(call_info_fields))
+
+    return '\t'.join(calls_values)
+
+  def _variant_to_vcf_line(self, variant):
+    """Converts the variant back to a line that can be written to a .vcf."""
+    info_field = self._get_variant_info(variant)
+    format_field = self._get_variant_format(variant)
+    calls_field = self._get_variant_calls(variant)
+
+    columns = [
+        variant.reference_name,
+        variant.start + 1 if variant.start else None,
+        ';'.join(variant.names),
+        variant.reference_bases,
+        ','.join(variant.alternate_bases),
+        variant.quality,
+        ';'.join(variant.filters),
+        info_field,
+        format_field,
+    ]
+    if calls_field:
+      columns.append(calls_field)
+
+    return '\t'.join(
+        [str(c) if c else MISSING_FIELD_VALUE for c in columns]) + '\n'
```

Comment:
Should check with isNoneOrEmpty here as well.

Reply:
Done.